### PR TITLE
Build EditorExtension as part of CI job

### LINKED ISSUE #2070: Build `SwiftRefactorExtension.xcodeproj` in CI
We should ensure that `EditorExtension/SwiftRefactorExtension.xcodeproj` builds in CI by building it when running `swift-syntax-dev-utils build`.

### PR DESCRIPTION
**Pull request by kimdv**
Resolves #2070

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

File: SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/BuildCommand.swift [hunk 1]
```diff
@@ -79,6 +79,26 @@ extension BuildCommand {
     return result.stdout
   }
 
+  func invokeXcodeBuild(projectPath: URL) throws -> ProcessResult {
+    guard let xcodebuildExec = Paths.xcodebuildExec else {
+      throw ScriptExectutionError(
+        message: """
+          Error: Could not find xcbuild.
```
Comment (ahoppen) [inline SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/BuildCommand.swift:86]:
```suggestion
          Error: Could not find xcodebuild.
```

File: SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/Build.swift [hunk 2]
```diff
@@ -20,5 +20,11 @@ struct Build: ParsableCommand, BuildCommand {
   func run() throws {
     try buildTarget(packageDir: Paths.packageDir, targetName: "SwiftSyntax-all")
     try buildTarget(packageDir: Paths.examplesDir, targetName: "Examples-all")
+    try buildEditorExtension()
+  }
+
+  private func buildEditorExtension() throws {
+    logSection("Building Edittor Extension")
```
Comment (ahoppen) [inline SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/Build.swift:27]:
```suggestion
    logSection("Building Editor Extension")
```

**Review by ahoppen — APPROVED** [review]
A couple typos, otherwise LGTM. Nice to review these changes in Swift instead of Python.

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
We should only build the editor extension on macOS because there’s no xcodebuild on Linux.

**Comment by kimdv** [comment]
> We should only build the editor extension on macOS because there’s no xcodebuild on Linux.

Just fixed it. Triggering CI

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**